File: tests/fixtures/java_sample/src/main/java/com/example/UserService.java
package com.example;

import java.util.List;
import java.util.HashMap;

public class UserService {
    public static final String VERSION = "1.0";
    private static final int MAX_SIZE = 100;

    private final String name;

    public UserService(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    private static int count(List<?> items) {
        return items.size();
    }

    public static UserService create(String n) {
        return new UserService(n);
    }
}

public interface Processor {
    void process(String input);

    default String preprocess(String s) {
        return s.trim();
    }
}

public enum Status {
    ACTIVE, INACTIVE, PENDING;

    public boolean isActive() {
        return this == ACTIVE;
    }
}

@interface MyAnnotation {
    String value();
    int timeout() default 30;
}

abstract class AbstractBase {
    abstract void doWork();
}

File: tests/fixtures/kotlin_sample/src/main/kotlin/com/example/UserService.kt
package com.example

import java.util.List
import java.util.HashMap

class UserService(val name: String) {
    companion object {
        const val VERSION = "1.0"
    }

    fun getName(): String = name

    private fun count(items: List<*>): Int = items.size

    fun create(n: String): UserService = UserService(n)
}

interface Processor {
    fun process(input: String)
    fun preprocess(s: String): String = s.trim()
}

enum class Status {
    ACTIVE, INACTIVE, PENDING;

    fun isActive(): Boolean = this == ACTIVE
}

object Singleton {
    fun getInstance(): Singleton = this
}

data class User(val id: Int, val name: String)

fun topLevelFn(): Unit = Unit
